frontend/src/components/resourceMap/graph/graphLayout.tsx

```diff
@@ -0,0 +1,246 @@
+import { Edge, EdgeMarker, Node } from '@xyflow/react';
+import { ElkExtendedEdge, ElkNode } from 'elkjs';
+import ELK from 'elkjs';
+import { forEachNode, GraphNode, KubeObjectNode } from './graphModel';
+
+type ElkNodeWithData = Omit<ElkNode, 'edges'> & {
+  type: string;
+  data: any;
+  edges?: ElkEdgeWithData[];
+};
+
+type ElkEdgeWithData = ElkExtendedEdge & {
+  type: string;
+  data: any;
+};
+
+const elk = new ELK({
+  defaultLayoutOptions: {},
+});
+
+const layoutOptions = {
+  nodeSize: {
+    width: 220,
+    height: 70,
+  },
+};
+
+/**
+ * To increase readability of the graph we can sort nodes left-to-right
+ * Where more 'owner' nodes like Deployment or ReplicaSet are on the left
+ */
+function getPartitionLayer(node: KubeObjectNode) {
+  let layer: any = undefined;
+
+  if ('resource' in node.data) {
+    const res = node.data.resource;
+    if (res.kind === 'Deployment') {
```

Comment:
How about?

```ts
const partitionLayers = [
  [
    'Deployment',
  ],
  [
    'ReplicaSet',
    'ServiceAccount',
  ],
  ...
];

if ('resource' in node.data) {
  const kind = node.data.resource.kind;
  const layer = partitionLayers.findIndex(partition => kind);

  if (layer !== -1) {
    return layer;
  }
}
```
If we could return a negative value to mean it wasn't found, then it'd be even better.

Reply:
a layer can have a negative value, meaning layout algorithm would place all those nodes on the left and we want lack of layer value so it can place it where it fits best.